scapy/fields.py

```diff
@@ -1215,7 +1241,7 @@ def any2i_one(self, pkt, x):
     def i2repr_one(self, pkt, x):
         if self not in conf.noenum and not isinstance(x, VolatileValue):
             try:
-                return self.i2s[x]
+                return self.i2s[x] if self.i2s[x] is not None else repr(x)
```

Comment:
Is that needed because of the ObservableDict? Or for another reason?

Reply:
Why don't you `del` the value rather than setting it to `None`? I think that's how it should be done, if I understand correctly.

Reply:
Sorry, I wasn't aware of this del operator. 
Changed the code.